SPARCore/GADArticle.m

```diff
@@ -51,57 +89,40 @@ - (void) fetchFulltextWithCompletion: (void(^_Nonnull)(GADRemoteModel * *_Nullab
-    NSError *error = nil;
-    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
-                                                         options:kNilOptions error:&error];
-    
-    if (error != nil) {
-        NSLog(@"Error parsing JSON.");
+    GADArticle *article = [[GADArticle alloc] init];
+    int timeStamp = (int)jsonDict[API_DATE_PUBLISHED];
+    article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
+    article.brief = jsonDict[API_BRIEF];
+    article.headerImage = [NSURL URLWithString:jsonDict[API_HEADER_IMAGE]];
+    article.publicationId = jsonDict[API_PUBLICATION_ID];
+    article.articleId = jsonDict[API_ARTICLE_ID];
+    article.title = jsonDict[API_TITLE];
+    article.authors = jsonDict[API_AUTHORS];
+    if (jsonDict[API_CONTENT]) {
+        article.content = jsonDict[API_CONTENT];
     }
     
-    for (NSDictionary *element in jsonArray) {
-        //Map fields of element to fields of article
-        GADArticle *article = [[GADArticle alloc] init];
-        //datePublished field is a UNIX Timestamp number - converting to NSDate here
-        int timeStamp = (int)element[API_DATE_PUBLISHED];
-        article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
-        article.brief = element[API_BRIEF];
-        article.headerImage = element[API_HEADER_IMAGE];
-        article.publicationId = element[API_PUBLICATION_ID];
-        article.articleId = element[API_ARTICLE_ID];
-        article.title = element[API_TITLE];
-        if (element[API_CONTENT]) {
-            article.content = element[API_CONTENT];
-            article.authors = element[API_AUTHORS];
-        }
-        [articles addObject:article];
-    }
-    return articles;
-}
-
-+ (NSURL *) baseURL {
-    NSURL *queryURL = [NSURL URLWithString:API_HOSTNAME];
-    queryURL = [NSURL URLWithString:API_PREFIX relativeToURL:queryURL];
-    return queryURL;
-}
-
-+ (NSURL *) urlForArticlesFromPublication: (NSString *)publicationId{
-    NSURL *queryURL = [GADArticle baseURL];
-    queryURL = [NSURL URLWithString:API_PUBLICATION_PATH relativeToURL:queryURL];
-    queryURL = [NSURL URLWithString:publicationId relativeToURL:queryURL];
-    queryURL = [NSURL URLWithString:API_ARTICLE_PATH relativeToURL:queryURL];
-    return queryURL;
+    NSMutableArray *wrapper = [NSMutableArray new];
+    [wrapper addObject:article];
+    
+    return wrapper;
 }
 
-+ (NSURL *) createURLWithArticle: (NSString *)articleId publication:(NSString *)publicationId{
-    NSURL *queryURL = [GADArticle urlForArticlesFromPublication:publicationId];
-    queryURL = [NSURL URLWithString:articleId relativeToURL:queryURL];
+- (NSURL *) urlForFullArticle{
+    GADPublication *pub = [GADPublication new];
+    pub.publicationId=self.publicationId;
+    NSURL *queryURL = [pub urlForArticles];
+    queryURL = [NSURL URLWithString:[self articleId] relativeToURL:queryURL];
     return queryURL;
 }
 
+// needs to be changed
```

Comment:
This is not a particularly helpful comment. Please make it more helpful or remove it.